compiler/qsc_data_structures/src/span.rs

```diff
@@ -16,6 +16,16 @@ pub struct Span {
     pub hi: u32,
 }
 
+impl Span {
+    #[must_use]
+    pub fn subtract(&self, offset: u32) -> Span {
```

Comment:
Since we already impl `Add<u32>` for `Span`, I suggest implementing `Sub<u32>` for symmetry (and also for the convenience of using the `-` operator)